app/Plugin.php

```diff
@@ -205,9 +202,15 @@ public function handleRemove() {
         // if installed, first rollback migrations and delete all files and presets
         if(isset($this->installed_at)) {
             $this->handleUninstall();
+            $this->rollbackMigrations();
+            $this->removePermissions();
+            $this->uninstallPresets();
         }
 
+        $this->removePreferences();
         sp_remove_dir(base_path("app/Plugins/$this->name"));
+
+        $this->delete();
```

Comment:
Great, now the migrations will be only rolled back if the plugin os permanently removed, right?

Reply:
Yes. I'm not sure if we should still remove permissions and presets. We'll see if this leads to unexpected behaviour with inactive plugins